query-connector/src/app/backend/dbServices/fhir-servers.ts

```diff
@@ -0,0 +1,295 @@
+"use server";
+import { Pool } from "pg";
+import { getDbClient } from "../dbClient";
+import { transaction } from "../../shared/decorators";
+import { FhirServerConfig } from "@/app/models/entities/fhir-servers";
+import { superAdminAccessCheck } from "@/app/utils/auth";
+
+class FhirServerConfigService {
+  private static dbClient: Pool = getDbClient();
+  private static cachedFhirServerConfigs: FhirServerConfig[] | null = null;
+
+  /**
+   * Fetches the configuration for a FHIR server from the database.
+   * @param forceRefresh - Whether to flush the config cache
+   * @returns The configuration for the FHIR server.
+   */
+  static async getFhirServerConfigs(forceRefresh = false) {
```

Comment:
Can you make sure [this code](https://github.com/CDCgov/dibbs-query-connector/blob/0d7eaceb388434284d3b646ffd73848631ad5c44/query-connector/src/app/backend/fhir-servers.ts#L289) gets migrated?. These are the methods that I was talking about where one needs to perform role checks and the other does not. 

The query page should be consuming getFhirServerConfigs and the fhir servers page getFhirServersList. Right now in this branch the query page has runtime errors because of the role access checks.